Invalid memory address or nil pointer dereference when running through CI/CD

My root terragrunt.hcl file was like this 
```
..................
# Generate an AWS provider block.
generate "provider" {
  path      = "provider.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
provider "aws" {
  region = "${local.region}"
}
EOF
}
...............
```

The only change I did was adding another generate block, as below -
```
........................
generate "provider" {
  path      = "provider.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
provider "aws" {
  region = "${local.region}"
}
EOF
}

generate "provider" {
  path      = "remote.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
data "terraform_remote_state" "global" {
  backend = "s3"
  config = {
    bucket = "remote-state-bucket-b677a"
    key    = "path/to/terraform.tfstate"
    region = "us-west-2"
  }
}
EOF
}
........................
```

**Command I am running is -** 

_terragrunt run-all plan -input=false -out plan.out --terragrunt-non-interactive --terragrunt-include-dir "/us-west-2/pro/maintenance" --terragrunt-working-dir "myawsaccountalias"_

**This command is working perfectly fine while i run interactively through Powershell/normal command line.**

While running in azure devops powershell task, it fails with below error - 

```
Starting: Terragrunt plan
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.190.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent1\_work\_temp\1947a2b7-60ca-41cf-92cc-88f1b7b609ec.ps1'"

time=2021-10-14T06:45:24Z level=info msg=Stack at hpsa-backend-pro:
  => Module C:/agent1/_work/5/s/account-alias/us-west-2/pro/appes (excluded: true, dependencies: [C:/agent1/_work/5/s/account-alias/us-west-2/pro/maintenance])
  => Module C:/agent1/_work/5/s/account-alias/us-west-2/pro/maintenance (excluded: false, dependencies: [])
  => Module C:/agent1/_work/5/s/account-alias/us-west-2/pro/bitapp (excluded: true, dependencies: [C:/agent1/_work/5/s/account-alias/us-west-2/pro/maintenance])
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x167fb61]

goroutine 19 [running]:
github.com/gruntwork-io/terragrunt/config.ParseConfigString(0xc000290000, 0x5d9, 0xc0007a41e0, 0x0, 0xc00053d130, 0x4d, 0x0, 0x17c6800, 0xffffffffffffff01, 0xc0004f2670)
	/home/circleci/project/config/config.go:650 +0x321
github.com/gruntwork-io/terragrunt/config.ParseConfigFile(0xc00053d130, 0x4d, 0xc0007a41e0, 0x0, 0x0, 0x1, 0x1, 0x0)
	/home/circleci/project/config/config.go:554 +0xa9
github.com/gruntwork-io/terragrunt/config.ReadTerragruntConfig(0xc0007a41e0, 0x0, 0x0, 0x19db7d3)
	/home/circleci/project/config/config.go:543 +0xd8
github.com/gruntwork-io/terragrunt/cli.RunTerragrunt(0xc0007a41e0, 0x5, 0x19dfb73)
	/home/circleci/project/cli/cli_app.go:368 +0x12c
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runNow(0xc00028d130, 0xc000093f5f, 0x0)
	/home/circleci/project/configstack/running_module.go:239 +0x18d
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runModuleWhenReady(0xc00028d130, 0xc0000382a0)
	/home/circleci/project/configstack/running_module.go:202 +0xde
github.com/gruntwork-io/terragrunt/configstack.runModules.func1(0xc0002243e0, 0xc0000382a0, 0xc00028d130)
	/home/circleci/project/configstack/running_module.go:172 +0x65
created by github.com/gruntwork-io/terragrunt/configstack.runModules
	/home/circleci/project/configstack/running_module.go:170 +0x11b
##[error]PowerShell exited with code '1'.
Finishing: Terragrunt plan

```

--- a/config/config.go
+++ b/config/config.go
@@ -661,9 +661,12 @@ func ParseConfigString(
 	// If this file includes another, parse and merge it.  Otherwise just return this config.
 	if trackInclude != nil {
 		config, err := handleInclude(config, trackInclude, terragruntOptions, contextExtensions.DecodedDependencies)
+		if err != nil {
+			return nil, err
+		}
 		// Saving processed includes into configuration, direct assignment since nested includes aren't supported
 		config.ProcessedIncludes = trackInclude.CurrentMap
-		return config, err
+		return config, nil
 	}
 	return config, nil
 }

